docs/admin/out-of-resource.md

```diff
@@ -151,6 +175,76 @@ and the node only has `Guaranteed` pod(s) remaining, then the node must choose t
 `Guaranteed` pod in order to preserve node stability, and to limit the impact
 of the unexpected consumption to other `Guaranteed` pod(s).
 
+### Disk based evictions
+
+The `kubelet` responds differently to disk based eviction thresholds if the machine has a
+dedicatd `imagefs` configured for the container runtime.
+
+#### With Imagefs
+
+If `nodefs` filesystem has met eviction thresholds, `kubelet` will free up disk space in the following order:
+
+1. Delete dead pods/containers
+1. Evict Pods if required.
+
+If `imagefs` filesystem has met eviction thresholds, `kubelet` will free up disk space in the following order:
+
+1. Delete all unused images
+1. Evict Pods if required.
+
+#### Without Imagefs
+
+If `nodefs` filesystem has met eviction thresholds, `kubelet` will free up disk space in the following order:
+
+1. Delete dead pods/containers
+1. Delete all unused images
+1. Evict Pods if required.
+
+#### Ranking pods for eviction
+
+Local disk is a `BestEffort` resource.  If necessary, `kubelet` will evict pods one at a time to reclaim
+disk when `DiskPressure` is encountered.  The `kubelet` will rank pods by quality of service.  If the `kubelet`
+is responding to `inode` starvation, it will reclaim `inodes` by evicting pods with the lowest quality of service
+first.  If the `kubelet` is responding to lack of available disk, it will rank pods within a quality of service
+that consumes the largest amount of disk and kill those first.
+
+##### Without Imagefs
+
+If `nodefs` is triggering evictions, `kubelet` will sort pods based on their total disk usage
+- local volumes + logs & writable layer of all its containers.
+
+##### With Imagefs
+
+If `nodefs` is triggering evictions, `kubelet` will sort pods based on the usage on `nodefs`
+- local volumes + logs of all its containers.
+
+If `imagefs` is triggering evictions, `kubelet` will sort pods based on the writable layer usage of all its containers.
+
+##### Minimum eviction reclaim
```

Comment:
no, its supported for any resource under pressure, but it was new to 1.4 so got collapsed into this PR.  the heading depth confuses this, so i updated it, thanks.